Data Stream support for Audit- Log (#8356)
opensearch-project/documentation-website: _security/audit-logs/storage-types.md

:--- | :---
debug | Outputs to stdout. Useful for testing and debugging.
internal_opensearch | Writes to an audit index on the current OpenSearch cluster.
internal_opensearch_data_stream | Writes to an audit log data stream on the current OpenSearch cluster.
external_opensearch | Writes to an audit index on a remote OpenSearch cluster.
webhook | Sends events to an arbitrary HTTP endpoint.
log4j | Writes the events to a Log4j logger. You can use any Log4j [appender](https://logging.apache.org/log4j/2.x/manual/appenders.html), such as SNMP, JDBC, Cassandra, and Kafka.

You configure the output location in `opensearch.yml`:

```
plugins.security.audit.type: <debug|internal_opensearch|internal_opensearch_data_stream|external_opensearch|webhook|log4j>
```

`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` can be customized with additional configuration options. For more information, see [Internal OpenSearch data streams](#internal-opensearch-data-streams).

## Internal OpenSearch data streams

> You can configure the `internal_opensearch_data_stream` type with the following configuration parameters.

Reviewer: ```suggestion
You can configure the `internal_opensearch_data_stream` type with the following parameters.
```

:--- | :---
debug | Outputs to stdout. Useful for testing and debugging.
internal_opensearch | Writes to an audit index on the current OpenSearch cluster.
internal_opensearch_data_stream | Writes to an audit Data Stream on the current OpenSearch cluster.
external_opensearch | Writes to an audit index on a remote OpenSearch cluster.
webhook | Sends events to an arbitrary HTTP endpoint.
log4j | Writes the events to a Log4j logger. You can use any Log4j [appender](https://logging.apache.org/log4j/2.x/manual/appenders.html), such as SNMP, JDBC, Cassandra, and Kafka.

You configure the output location in `opensearch.yml`:

```
plugins.security.audit.type: <debug|internal_opensearch|internal_opensearch_data_stream|external_opensearch|webhook|log4j>
```

> `internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` all have additional configuration options. Details follow.

Reviewer 2: ```suggestion
`internal_opensearch_data_stream`, `external_opensearch`, `webhook`, and `log4j` can be customized with additional configuration options. For more information, see [Internal OpenSearch data streams](#internal-opensearch-data-streams).
```

> ## Internal Opensearch Data Stream

Reviewer 2: ```suggestion
## Internal OpenSearch data streams
```

> The type `internal_opensearch_data_stream` have some additional configuration paramters.

Reviewer 2: ```suggestion
You can configure the `internal_opensearch_data_stream` type with the following configuration parameters.
```

> ### Basics settings

Reviewer 2: ```suggestion
### Basic settings
```

> ### Basic settings

Reviewer 2: Will there be any additional settings called out in this section in the future?

> ### Basic settings

Reviewer 2: ```suggestion

```

Name | Data type | Description
:--- | :--- | :---

> `plugins.security.audit.config.data_stream.name` | String | Name of the auditlog Data Stream. The Default is `opensearch-security-auditlog`.

Reviewer 2: ```suggestion
`plugins.security.audit.config.data_stream.name` | String | The name of the audit log data stream. Default is `opensearch-security-auditlog`.
```

### Template settings

Name | Data type | Description
:--- | :--- | :---

> `plugins.security.audit.config.data_stream.template.manage` | Boolean | If enabled, the template for the Data Stream is managed by Opensearch. The Default is `true`.

Reviewer 2: ```suggestion
`plugins.security.audit.config.data_stream.template.manage` | Boolean | If `true`, the template for the data stream is managed by Opensearch. Default is `true`.
```

> `plugins.security.audit.config.data_stream.template.manage` | Boolean | If `true`, the template for the data stream is managed by Opensearch. Default is `true`.

Reviewer 2: ```suggestion
`plugins.security.audit.config.data_stream.template.manage` | Boolean | When `true`, the template for the data stream is managed by Opensearch. Default is `true`.
```

> `plugins.security.audit.config.data_stream.template.name` | String | Name of the template. The Default is `opensearch-security-auditlog`.

Reviewer 2: ```suggestion
`plugins.security.audit.config.data_stream.template.name` | String | The name of the data stream template. Default is `opensearch-security-auditlog`.
```

> `plugins.security.audit.config.data_stream.template.number_of_replicas` | Integer | Number of replicas for the Data Stream. The default is `0`.

Reviewer 2: ```suggestion
`plugins.security.audit.config.data_stream.template.number_of_replicas` | Integer | The number of replicas for the data stream. Default is `0`.
```

> `plugins.security.audit.config.data_stream.template.number_of_shards` | Integer | Number of Shards for the Data Stream. The default is `1`.

Reviewer 2: ```suggestion
`plugins.security.audit.config.data_stream.template.number_of_shards` | Integer | The number of shards for the data stream. Default is `1`.
```
